controller.go

```diff
@@ -931,15 +932,15 @@ func parseAllocateConstraintsResponse(source interface{}, machine *machine) (Con
 	constraintsMap := valid["constraints_by_type"].(map[string]interface{})
 	result := ConstraintMatches{
 		Interfaces: make(map[string][]Interface),
-		Storage:    make(map[string][]BlockDevice),
+		Storage:    make(map[string][]StorageDevice),
 	}
 
 	if interfaceMatches, found := constraintsMap["interfaces"]; found {
 		matches := convertConstraintMatches(interfaceMatches)
 		for label, ids := range matches {
 			interfaces := make([]Interface, len(ids))
 			for index, id := range ids {
-				iface := machine.Interface(id)
+				iface := machine.Interface(id.(int))
```

Comment:
This is dangerous given the change above to schema.Any() as it could panic
Why was the change to schema.Any() needed?